### PR TITLE
Backfill missing reward for issue 763

### PR DESCRIPTION
**Pull request by Index0011**
Add function to backfill missing reward due to #763

### CODE REVIEW

File: pallets/phala/src/stakepool.rs [hunk 1]
```diff
@@ -541,6 +543,32 @@ pub mod pallet {
...
+		/// manually assign rewards to pools by root for fixing issue 763
+		///
+		///Reuires:
+		/// 1. The caller is root
+		/// 2. assigned pool must currently exist
+		/// 3. reeward is positive
+		#[pallet::weight(0)]
+		pub fn force_assign_reward(
+			origin: OriginFor<T>,
+			pid: u64,
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:555]: Better to make the args a Vec to allow batch operation

**Review by h4x3rotab — COMMENTED** [review]
Almost LGTM. We may need to add unit test coverage

File: pallets/phala/src/stakepool.rs [hunk 2]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// manually assign rewards to pools by root for fixing issue 763
+		///
+		///Reuires:
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:548]:
```suggestion
		/// Requires:
```

File: pallets/phala/src/stakepool.rs [hunk 3]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// manually assign rewards to pools by root for fixing issue 763
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:546]: Upper case just like regular english sentences. Also, it's not exactly "by root", rather the `MiningSwitchOrigin`.

File: pallets/phala/src/stakepool.rs [hunk 4]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
...
+		///Reuires:
+		/// 1. The caller is root
+		/// 2. assigned pool must currently exist
+		/// 3. reeward is positive
+		#[pallet::weight(0)]
+		pub fn force_assign_reward(
+			origin: OriginFor<T>,
+			reward_arr: Vec<(u64, BalanceOf<T>)>,
+		) -> DispatchResult {
+			// origin must be root
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:557]:
```suggestion
			// Origin must be MiningSwitchOrigin
```

File: pallets/phala/src/stakepool.rs [hunk 5]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
...
+			T::MiningSwitchOrigin::ensure_origin(origin)?;
+			for pair in reward_arr {
+			    let (pid, reward) = pair;
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:560]:
```suggestion
			for (pid, reward) in reward_arr {
```

File: pallets/phala/src/stakepool.rs [hunk 6]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
...
+			    //assigned pool must exist
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:561]:
```suggestion
			    // The assigned pool must exist
```

Space after `//`; upper case

File: pallets/phala/src/stakepool.rs [hunk 7]
```diff
@@ -541,6 +543,35 @@ pub mod pallet {
...
+			    let mut pool_info = Self::ensure_pool(pid)?;
+			    //reward must be positive
+			    if reward <= Zero::zero() {
+				continue
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:565]: indent. Maybe you need to run `cargo fmt`

File: pallets/phala/src/stakepool.rs [hunk 8]
```diff
@@ -554,8 +554,9 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			reward_arr: Vec<(u64, BalanceOf<T>)>,
 		) -> DispatchResult {
-			// origin must be root
+			// origin must be miningSwitchOrigin
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:557]:
```suggestion
			// Origin must be MiningSwitchOrigin
```

File: pallets/phala/src/stakepool.rs (hunk 6 above)
Comment (h4x3rotab) [reply pallets/phala/src/stakepool.rs:562]: Please fix

File: pallets/phala/src/stakepool.rs (hunk 5 above)
Comment (h4x3rotab) [reply pallets/phala/src/stakepool.rs:561]: Please fix

File: pallets/phala/src/stakepool.rs (hunk 7 above)
Comment (h4x3rotab) [reply pallets/phala/src/stakepool.rs:566]: Please fix

File: pallets/phala/src/stakepool.rs [hunk 9]
```diff
@@ -554,12 +554,13 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			reward_arr: Vec<(u64, BalanceOf<T>)>,
 		) -> DispatchResult {
-			// origin must be miningSwitchOrigin
+
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:557]: Remove empty space

File: pallets/phala/src/stakepool.rs [hunk 10]
```diff
@@ -554,12 +554,13 @@ pub mod pallet {
...
+			// Origin must be MiningSwitchOrigin
 			T::MiningSwitchOrigin::ensure_origin(origin)?;
 
-			for pair in reward_arr {
-			    let (pid, reward) = pair;
-			    //assigned pool must exist
+			for (pid, reward) in reward_arr {
+			    //The assigned pool must exist
+
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:563]: Why space?

File: pallets/phala/src/stakepool.rs [hunk 11]
```diff
@@ -554,12 +554,13 @@ pub mod pallet {
...
 			    let mut pool_info = Self::ensure_pool(pid)?;
 			    //reward must be positive
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:565]: Space, capitalize

File: pallets/phala/src/stakepool.rs [hunk 12]
```diff
@@ -554,12 +554,13 @@ pub mod pallet {
...
-			// origin must be miningSwitchOrigin
+
+			// Origin must be MiningSwitchOrigin
 			T::MiningSwitchOrigin::ensure_origin(origin)?;
 
-			for pair in reward_arr {
-			    let (pid, reward) = pair;
-			    //assigned pool must exist
+			for (pid, reward) in reward_arr {
+			    //The assigned pool must exist
```
Comment (h4x3rotab) [inline pallets/phala/src/stakepool.rs:562]:
```suggestion
			    // The assigned pool must exist
```

File: pallets/phala/src/stakepool.rs (hunk 9 above)
Comment (Index0011) [reply pallets/phala/src/stakepool.rs]: done

File: pallets/phala/src/stakepool.rs (hunk 11 above)
Comment (Index0011) [reply pallets/phala/src/stakepool.rs]: ack

File: pallets/phala/src/stakepool.rs (hunk 7 above)
Comment (h4x3rotab) [reply pallets/phala/src/stakepool.rs:564]: Not fixed yet.

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**